pithos/pithos.py

```diff
@@ -843,7 +845,18 @@ def update_song_row(self, song = None):
         if song:
             self.songs_model[song.index][1] = self.song_text(song)
             self.songs_model[song.index][2] = self.song_icon(song) or ""
-        return self.playing
+        if self.playing:
+            return True
+
+    def create_ui_loop(self):
+        if self.ui_loop_timer_id is not None:
```

Comment:
self.ui_loop_timer_id has to be reset somehow(to None or 0 or whatever) otherwise when the loop dies it won't be able to be restarted since the ID is not cleared.

Could be as easy as - 

if not self.playing:
    self.ui_loop_timer_id = None (or 0 or whatever)
return return self.playing

in update_song_row